presto-main/src/main/java/com/facebook/presto/operator/aggregation/MapUnionAggregation.java

```diff
@@ -0,0 +1,129 @@
+    private static final MethodHandle COMBINE_FUNCTION = methodHandle(MapUnionAggregation.class, "combine", KeyValuePairsState.class, KeyValuePairsState.class);
+
+    public MapUnionAggregation()
+    {
+        super(NAME, ImmutableList.of(comparableTypeParameter("K"), typeVariable("V")), ImmutableList.of(), "map<K,V>", ImmutableList.of("map<K,V>"));
+    }
+
+    @Override
+    public String getDescription()
+    {
+        return "Aggregate all the maps into a single map";
+    }
+
+    @Override
+    public InternalAggregationFunction specialize(BoundVariables boundVariables, int arity, TypeManager typeManager, FunctionRegistry functionRegistry)
+    {
+        Type keyType = boundVariables.getTypeVariable("K");
+        Type valueType = boundVariables.getTypeVariable("V");
+        return generateAggregation(keyType, valueType);
+    }
+
+    private static InternalAggregationFunction generateAggregation(Type keyType, Type valueType)
+    {
+        DynamicClassLoader classLoader = new DynamicClassLoader(MapUnionAggregation.class.getClassLoader());
+        Type outputType = new MapType(keyType, valueType);
+        List<Type> inputTypes = ImmutableList.of(outputType);
+        KeyValuePairStateSerializer stateSerializer = new KeyValuePairStateSerializer(keyType, valueType, false);
+        Type intermediateType = stateSerializer.getSerializedType();
+
+        AggregationMetadata metadata = new AggregationMetadata(
+                generateAggregationName(NAME, outputType, inputTypes),
+                createInputParameterMetadata(outputType),
+                INPUT_FUNCTION.bindTo(keyType).bindTo(valueType),
+                null,
+                null,
+                COMBINE_FUNCTION,
+                OUTPUT_FUNCTION,
+                KeyValuePairsState.class,
+                stateSerializer,
+                new KeyValuePairsStateFactory(keyType, valueType),
+                outputType,
+                false);
+
+        GenericAccumulatorFactoryBinder factory = new AccumulatorCompiler().generateAccumulatorFactoryBinder(metadata, classLoader);
+        return new InternalAggregationFunction(NAME, inputTypes, intermediateType, outputType, true, false, factory);
+    }
+
+    private static List<ParameterMetadata> createInputParameterMetadata(Type inputType)
+    {
+        return ImmutableList.of(
+                new ParameterMetadata(STATE),
+                new ParameterMetadata(NULLABLE_BLOCK_INPUT_CHANNEL, inputType),
+                new ParameterMetadata(BLOCK_INDEX));
+    }
+
+    public static void input(Type keyType, Type valueType, KeyValuePairsState state, Block value, int position)
+    {
+        KeyValuePairs pairs = state.get();
+        if (pairs == null) {
+            pairs = new KeyValuePairs(keyType, valueType);
+            state.set(pairs);
+        }
+
+        long startSize = pairs.estimatedInMemorySize();
+        Block entryBlock = value.getObject(position, Block.class);
```

Comment:
This should be using `MapType.getObject`. Calling `Block.get*` outside of a `Type` is discouraged